irods/collection.py

```diff
@@ -78,5 +85,14 @@ def walk(self, topdown=True):
         if not topdown:
             yield (self, self.subcollections, self.data_objects)
 
+    @staticmethod
+    def normalize_path(path):
+        path_ = _minimize_slashes(path).rstrip('/')
```

Comment:
should this also handle... dot (`.`) and dot-dot (`..`)?  probably should be a separate issue if so.

@korydraughn do we already have a way to do this in the server - either that can be consulted / called / emulated?

Reply:
oh... i see something similar being tested below... ignore if this is already handled.

Reply:
I think I've handled the ` . ` and ` .. ` now, but will write a test for that too.

Reply:
I could still make `.` and `..` a separate issue.  (Related: session `chdir`, `cwd`,  and other "enhancements"  that are actually extras. (The new `iRODSPath` class is not extraneous - it performs the path normalization in direct support of this issue, #323)

Reply:
yeah, maybe . and .. should be separately described/handled in a new issue.

Reply:
#344 created.